subprojects/messaging/src/main/java/org/gradle/internal/remote/internal/inet/SocketConnection.java

```diff
@@ -156,7 +156,7 @@ public SocketInputStream(SocketChannel socket) throws IOException {
             selector = Selector.open();
             socket.register(selector, SelectionKey.OP_READ);
             buffer = ByteBuffer.allocateDirect(4096);
-            buffer.limit(0);
+            ((Buffer)buffer).limit(0);
```

Comment:
This is to avoid some `NoSuchMethodException` when code compiled by Java 9 runs on Java 8.

Reply:
Cool, I was wondering what did cause the failure.